flexget/plugins/output/slack.py

```diff
@@ -0,0 +1,120 @@
+from __future__ import unicode_literals, division, absolute_import
+import logging
+
+from flexget import plugin
+from flexget.event import event
+from flexget.utils import json
+from flexget.utils.template import RenderError
+
+log = logging.getLogger('slack')
+
+
+class OutputSlack(object):
+    """
+    Example:
+
+      slack:
+        webhook-url: <string>
+        [text: <string>] (default: "{{task}} - Download started: {{series_name}} {{series_id}}" -- accepts Jinja2)
+        [channel: <string>] (override channel, use "@username" or "#channel")
+        [username: <string>] (override username)
+        [icon-emoji: <string>] (override emoji icon, do not include the colons: e.g., use "tv" instead of ":tv:")
+
+    Configuration parameters are also supported from entries (e.g., through set).
+    """
+    default_body = ('{{task}} - Download started: '
+                    '{% if series_name is defined %}'
+                    '{{tvdb_series_name|d(series_name)}} {{series_id}} {{tvdb_ep_name|d('')}}'
+                    '{% elif imdb_name is defined %}'
+                    '{{imdb_name}} {{imdb_year}}'
+                    '{% else %}'
+                    '{{title}}'
+                    '{% endif %}')
+    schema = {
+        'type': 'object',
+        'properties': {
+            'webhook-url': {'type': 'string'},
+            'text': {'type': 'string', 'default': default_body},
+            'channel': {'type': 'string'},
+            'username': {'type': 'string'},
+            'icon-emoji': {'type': 'string'},
+        },
+        'required': ['webhook-url'],
+        'additionalProperties': False
+    }
+
+    @plugin.priority(0)
+    def on_task_output(self, task, config):
+
+        for entry in task.accepted:
+            webhook_url = config.get('webhook-url')
```

Comment:
I thought about that, too.... From what I can tell, the first random string in the URL is the team, the 2nd is the service, and the 3rd is the key, which gets regenerated whenever the user clicks "regenerate" in the Slack configuration. I could break this out into three separate values and make them all required and then hard-code the rest of the URL, but I would have to add a hefty bit of explanation because the Slack website itself doesn't supply these individually, it simply gives the entire URL to copy and paste.